🚀 Feature: Don't require version if package has `"private": true`

### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

I have disabled the `package-json/valid-version` rule because it warns on private packages. Tools like `changeset` require that you remove the private package version number (instead of, say providing a useless one like 0.0.1) or it will try to manage that version, even if the package is private. Ideally this package would just allow no versions on private packages.

Thank you for making this package btw. It's great!

### Additional Info

_No response_

Sorry, I misspoke: it's the `valid-package-def` rule that is erroring when a package has a private specifier and no version:

```
/Volumes/sourcecode/npm-packages/packages/eslint-config-altano/package.json
  1:1  error  Missing required field: version  package-json/valid-package-def
```

This test case in valid-package-def.test.ts would reveal it:

```
		{
			code: `{
  "name": "panda",
  "private": true
}`,
			filename: "package.json",
		},
```

I can submit a PR if you'd like. I would create an exception for this error and only pass the `Missing required field: version` error through if `private: true` is not in the package? (I don't see a way of configuring package validator to use that logic itself). Would I somehow traverse the ast to determine it has a name, or ?

--- a/package.json
+++ b/package.json
@@ -60,7 +60,7 @@
 	"dependencies": {
 		"detect-indent": "6.1.0",
 		"detect-newline": "3.1.0",
-		"package-json-validator": "^0.6.3",
+		"package-json-validator": "^0.6.5",
 		"semver": "^7.5.4",
 		"sort-object-keys": "^1.1.3",
 		"sort-package-json": "^1.57.0",

